spec/memo_wise_spec.rb

```diff
@@ -785,4 +959,52 @@ def no_args
       end
     end
   end
+
+  describe "private APIs" do
+    describe ".method_visibility" do
+      subject { described_class.method_visibility(String, method_name) }
+
+      context "when method_name not a method on klass" do
+        let(:method_name) { :not_a_method }
+
+        it { expect { subject }.to raise_error(ArgumentError) }
+      end
+    end
+
+    describe ".original_class_from_singleton" do
+      subject { described_class.original_class_from_singleton(klass) }
+
+      context "when klass is not a singleton class" do
+        let(:klass) { String }
+
+        it { expect { subject }.to raise_error(ArgumentError) }
+      end
+
+      context "when klass is a singleton class of an original class" do
+        let(:klass) { original_class.singleton_class }
+
+        context "when assigned to a constant (normal case)" do
+          let(:original_class) { String }
+
+          it { is_expected.to eq(original_class) }
+        end
+
+        context "when not assigned to a constant (anonymous case)" do
```

Comment:
Wait can't we remove and reorganize the tests here, since there is only one case now?